File: tecnicell.client/src/app/Interfaces/business/ApiResponses/BatteryResponse.ts
import { BatteryHistory } from "../Models/Battery"
import { Image } from "../Models/Image";

export interface BatteryResponse {
    view:      BatteryView;
    image?:     Image;
    histories: BatteryHistory[];
}
export interface BatteryView {
    code:      string;
    type:      string;
    name:      string;
    salePrice: number;
    warranty:number;
    imageCode: string;
    quantity:  number;
    available: string;
}

File: Tecnicell.Server/Models/Responses/BatteryResponse.cs
using Tecnicell.Server.Models.Entity;
using Tecnicell.Server.Models.ViewModel.Phone;
using Tecnicell.Server.Models.ViewModel;
using Tecnicell.Server.Models.ViewModel.Battery;

namespace Tecnicell.Server.Models.Responses
{
    public class BatteryResponse
    {
        public BatteryView? View { get; set; }
        public ImageViewModel? Image { get; set; }
        public IEnumerable<BatteryHistoryViewModel> Histories { get; set; } = new List<BatteryHistoryViewModel>();
    }
}
